core/src/pcap.rs

```diff
@@ -0,0 +1,200 @@
+use crate::dpdk::{CoreId, PortId};
+use crate::ffi::{self, ToCString, ToResult};
+use crate::packets::Packet;
+use crate::{debug, Result};
+use libc;
+use std::fmt;
+use std::os::raw;
+use std::ptr::NonNull;
+
+/// DLT_EN10MB; LINKTYPE_ETHERNET=1; 10MB is historical
+const DLT_EN10MB: raw::c_int = 1;
+const PCAP_SNAPSHOT_LEN: raw::c_int = ffi::RTE_MBUF_DEFAULT_BUF_SIZE as raw::c_int;
+
+/// Pcap Writer for packets
+#[derive(Debug)]
+pub struct Pcap {
+    path: String,
+    handle: NonNull<ffi::pcap_t>,
+    dumper: NonNull<ffi::pcap_dumper_t>,
+}
+
+impl Pcap {
+    /// Create file for dumping packets into from given file a path.
+    pub fn create(path: &str) -> Result<Pcap> {
+        unsafe {
+            let handle = ffi::pcap_open_dead(DLT_EN10MB, PCAP_SNAPSHOT_LEN).to_result()?;
+            let dumper = ffi::pcap_dump_open(handle.as_ptr(), path.to_cstring().as_ptr())
+                .to_result()
+                .map_err(|err| {
+                    ffi::pcap_close(handle.as_ptr());
+                    err
+                })?;
+
+            debug!("PCAP file {} created", path);
+
+            Ok(Pcap {
+                path: path.to_string(),
+                handle,
+                dumper,
+            })
+        }
+    }
+
+    /// Append to already-existing file for dumping packets into from a given
+    /// file path.
+    pub fn append(path: &str) -> Result<Pcap> {
+        unsafe {
+            let handle = ffi::pcap_open_dead(DLT_EN10MB, PCAP_SNAPSHOT_LEN).to_result()?;
+            let dumper = ffi::pcap_dump_open_append(handle.as_ptr(), path.to_cstring().as_ptr())
+                .to_result()
+                .map_err(|err| {
+                    ffi::pcap_close(handle.as_ptr());
+                    err
+                })?;
+
+            Ok(Pcap {
+                path: path.to_string(),
+                handle,
+                dumper,
+            })
+        }
+    }
+
+    /// Write packets to PCAP file handler
+    pub fn write_packets<T: Packet + fmt::Debug>(&self, packets: &[T]) {
+        packets.iter().for_each(|p| {
+            if let Err(err) = self.write(p) {
+                debug!(
```

Comment:
not sure if the debug print is valuable. I'm thinking just let the error bubble up to a higher level.